src/com/company/Main.java

```diff
@@ -237,31 +234,34 @@ public static char readSign() {
         return inputText.toUpperCase().charAt(0);
     }
 
-    public static void drawField(int rowCount, int[][] tab2) {
+    public static void drawField(int boardSize, final PlayerSignType[][] board) {
 
         System.out.println("");
-        String firstLine = "     ";     // 3 spacje na początku na kolumnę numerów wierszy
+        StringBuilder firstLine = new StringBuilder("     ");     // 3 spacje na początku na kolumnę numerów wierszy
 
-        for (int i = 0; i < rowCount; i++) firstLine += (char) ('a' + i) + " ! ";
-        System.out.println(firstLine);
+        for (int i = 0; i < boardSize; i++) {
+            firstLine.append((char) ('a' + i)).append(" ! ");
+        }
+        System.out.println(firstLine.toString());
 
-        String horizontalFullLine = "   ";
-        for (int i = 0; i < rowCount * 4 + 1; i++) horizontalFullLine += "-";
+        StringBuilder horizontalFullLine = new StringBuilder("   ");
+        for (int i = 0; i < boardSize * 4 + 1; i++) {
+            horizontalFullLine.append("-");
+        }
 
-        for (int i = 0; i < rowCount; i++) {
+        for (int i = 0; i < boardSize; i++) {
             System.out.println(horizontalFullLine);
             // linia z danymi z tabeli
-            String lineWithData = " " + (i + 1) + " |";
-            for (int j = 0; j < rowCount; j++) {
-                switch (tab2[i][j]) {
-                    case ItIsX:
-                        lineWithData += " X |";
-                        break;
-                    case ItIsO:
-                        lineWithData += " O |";
+            StringBuilder lineWithData = new StringBuilder(" ");
+            lineWithData.append(i + 1);
+            lineWithData.append(" |");
```

Comment:
dlaczego robisz to w dwóch liniach a nie w jednej?

Reply:
Też już przerobiłam, szukałam błędu i rozbijałam to co było podejrzane. W najnowszej wersji jest sklejone do jednego wiersza. Zastanawiam się czy powinnam tworzyć stałe zawierające na przykład spację i znak pipe, albo trzy spacje i pipe, Muszę nad tymi stringami jeszcze pomyśleć.